connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/JakartaEmailListener.java

```diff
@@ -0,0 +1,140 @@
+
+  public JakartaEmailListener(JakartaUtils jakartaUtils) {
+    this.jakartaUtils = jakartaUtils;
+  }
+
+  public static JakartaEmailListener create(JakartaUtils sessionFactory) {
+    return new JakartaEmailListener(sessionFactory);
+  }
+
+  public void startListener(InboundConnectorContext context) {
+    EmailProperties emailProperties = context.bindProperties(EmailProperties.class);
+    Authentication authentication = emailProperties.getAuthentication();
+    Session session =
+        this.jakartaUtils.createSession(
+            emailProperties.getData().getImapConfig(), emailProperties.getAuthentication());
+    try {
+      this.store = session.getStore();
+      this.imapFolders = new ArrayList<>();
+      this.idleManager = new IdleManager(session, this.executorService);
+
+      this.jakartaUtils.connectStore(this.store, authentication);
+      List<String> inboxes = createInboxList(emailProperties.getData().getFolderToListen());
+      for (String inbox : inboxes) {
+        IMAPFolder folder = (IMAPFolder) store.getFolder(inbox);
+        folder.open(Folder.READ_WRITE);
+        folder.addMessageCountListener(
+            new MessageCountListener() {
+              @Override
+              public void messagesAdded(MessageCountEvent e) {
+                processNewEvent(
+                    e,
+                    context,
+                    emailProperties.getData().isTriggerAdded(),
+                    emailProperties.getData().isMarkAsRead());
+              }
+
+              @Override
+              public void messagesRemoved(MessageCountEvent e) {
+                processNewEvent(
+                    e,
+                    context,
+                    emailProperties.getData().isTriggerRemoved(),
+                    emailProperties.getData().isMarkAsRead());
+              }
+            });
+        this.imapFolders.add(folder);
+        idleManager.watch(folder);
+      }
+    } catch (MessagingException | IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void processNewEvent(
+      MessageCountEvent e,
+      InboundConnectorContext connectorContext,
+      boolean triggerAdded,
+      boolean markAsRead) {
+    IMAPFolder imapFolder = (IMAPFolder) e.getSource();
+    if (triggerAdded) {
+      Arrays.stream(e.getMessages())
+          .peek(
+              message -> {
+                if (markAsRead) this.jakartaUtils.markAsSeen(message);
+              })
+          .map(Email::createEmail)
+          .forEach(
+              email ->
+                  connectorContext.correlateWithResult(
+                      new ReadEmailResponse(
+                          email.getMessageId(),
+                          email.getFrom(),
+                          email.getSubject(),
+                          email.getSize(),
+                          email.getBody().getBodyAsPlainText(),
+                          email.getBody().getBodyAsHtml())));
```

Comment:
Could be extracted into a separate method and switched to a `map(this::correlateEmail)`

Reply:
Yes